lib/ddtrace/tracer.rb

```diff
@@ -325,7 +328,7 @@ def record(context)
     def record_context(context)
       trace = @context_flush.consume!(context)
 
-      write(trace) if trace && !trace.empty?
+      write(trace) if @enabled && trace && !trace.empty?
```

Comment:
Moved the `@enabled` check here to ease the testing around `tracer#write`. Before we'd had to double check if `tracer#write` was actually being called by also check `@enabled`.

Now we can safely assert on `tracer#write`, knowing that it is only called when the tracer actually wants to write a new trace.

Reply:
Interesting... it might be better to move this `enabled` behavior elsewhere such that we prevent spans from being recorded into the context in the first place. Not necessary to undertake right now, but perhaps when we improve disabling behavior.